src/sage/graphs/domination.py

```diff
@@ -338,6 +341,31 @@ def dominating_sets(g, k=1, independent=False, total=False,
         sage: next(g.dominating_sets())
         [1]
 
+    Minimum connected dominating sets of the Peterson graph::
+
+        sage: G = graphs.PetersenGraph()
+        sage: G.dominating_set(total=True, value_only=True)
+        4
+        sage: sorted(G.dominating_sets(k=1, connected=True))
+        [[0, 1, 2, 6],
+         [0, 1, 4, 5],
+         [0, 3, 4, 9],
+         [0, 5, 7, 8],
+         [1, 2, 3, 7],
+         [1, 6, 8, 9],
+         [2, 3, 4, 8],
+         [2, 5, 7, 9],
+         [3, 5, 6, 8],
+         [4, 6, 7, 9]]
+
```

Comment:
Add a doctest showing that the subgraph induced by a dominating set is connected. For instance:

```
sage: all(G.subgraph(vertices=dom).is_connected() for dom in G.dominating_set(k=1, connected=True))
True
```